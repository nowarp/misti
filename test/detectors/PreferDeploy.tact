contract A { init() {} }
contract B {
    owner: Address;
    init(owner: Address) { self.owner = owner; }
    receive() {
        let init = initOf A();
        send(SendParameters{ // Bad: Prefer deploy
            to: contractAddress(init),
            code: init.code,
            data: init.data,
            value: 0,
            body: beginCell().endCell(),
        });
        send(SendParameters{ // OK
            to: self.owner,
            value: 0,
            bounce: false,
            body: beginCell().endCell()
        });
    }
}
