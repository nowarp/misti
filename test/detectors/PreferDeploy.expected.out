[INFO] PreferDeploy: Prefer `deploy` over `send`
test/detectors/PreferDeploy.tact:7:9:
  6 |         let init = initOf A();
> 7 |         send(SendParameters{ // Bad: Prefer deploy
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  8 |             to: contractAddress(init),
Help: Use more gas-effecient `deploy` function: https://docs.tact-lang.org/ref/core-common/#deploy
See: https://nowarp.io/tools/misti/docs/detectors/PreferDeploy